### ISSUE TITLE
`i18n` function does not work in ExecuteAsTemplate

### ISSUE THREAD

**Issue by reuixiy** [gohugoio/hugo#7070]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
Hugo Static Site Generator v0.67.1/extended linux/amd64 BuildDate: unknown
</pre>

### Does this issue reproduce with the latest release?

Yes

---

[`script.html`](https://github.com/reuixiy/hugo-theme-meme/blob/f1f9876faed2d5241664605d796f3cc763b73818/layouts/partials/script.html#L8)

```html
{{- $copy := resources.Get "js/copy.js" | resources.ExecuteAsTemplate "js/copy-rendered.js" . -}}
```

[`copy.js`](https://github.com/reuixiy/hugo-theme-meme/blob/f1f9876faed2d5241664605d796f3cc763b73818/assets/js/copy.js#L7-L8)

```js
const copyText = '{{ i18n "copy" }}';
const copiedText = '{{ i18n "copied" }}';
```

Live demo: https://io-oi.me/en/about/ (with Hugo `v0.66.0`)

Hover the code block, the text should be [`Copy`](https://github.com/reuixiy/hugo-theme-meme/blob/f1f9876faed2d5241664605d796f3cc763b73818/i18n/en.toml#L23-L24) instead of [`复制`](https://github.com/reuixiy/hugo-theme-meme/blob/f1f9876faed2d5241664605d796f3cc763b73818/i18n/zh.toml#L23-L24).

Seems like a degradation? https://github.com/gohugoio/hugo/issues/6331

**Comment by bep** [comment]
I fixed a bug in 0.67.1. Now we use all the arguments to ExecuteAs... when we create the cache key. 

So:

```
 resources.ExecuteAsTemplate "js/copy-rendered.js" . 
```

There is the page (".") and the path (js/copy...).

The path is, however, language-agnostic (so to speak), so there is a "first win" situation here.

So, if you really want to create a JS per language, try:

```bash
...resources.ExecuteAsTemplate (printf "%s/js/copy-rendered.js" .Lang)  -}}
```

Also note that re my first comment above, it's not generally a great idea to pass the current page in as the context if you don't produce a "per page" resource (as you would execute the template x number of pages).

**Comment by reuixiy** [comment]
Thanks a lot!
